client/unsubscribe_transaction.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"fmt"
+	"strings"
+
+	msgs "github.com/energomonitor/bisquitt/messages"
+	"github.com/energomonitor/bisquitt/transactions"
+)
+
+type unsubscribeTransaction struct {
+	*transaction
+}
+
+func newUnsubscribeTransaction(client *Client, msgID uint16) *unsubscribeTransaction {
+	tLog := client.log.WithTag(fmt.Sprintf("UNSUBSCRIBE(%d)", msgID))
+	tLog.Debug("Created.")
+	return &unsubscribeTransaction{
+		transaction: &transaction{
+			RetryTransaction: transactions.NewRetryTransaction(
+				client.groupCtx, client.cfg.RetryDelay, client.cfg.RetryCount,
+				func(lastMsg interface{}) error {
+					tLog.Debug("Resend.")
+					return client.send(lastMsg.(msgs.Message))
+				},
+				func() {
+					tLog.Debug("Deleted.")
+					client.transactions.Delete(msgID)
+				},
+			),
+			client: client,
+			log:    tLog,
+		},
+	}
+}
+
+func (t *unsubscribeTransaction) Unsuback(_ *msgs.UnsubackMessage) {
+	var topicName string
+	unsubscribe := t.Data.(*msgs.UnsubscribeMessage)
+
+	switch unsubscribe.TopicIDType {
+	case msgs.TIT_STRING:
+		topicName = string(unsubscribe.TopicName)
+
+		t.log.Debug(`Topic "%s" unsubscribed`,
+			topicName,
+		)
```

Comment:
This debug log is fired on `TIT_STRING` unsubscribe only. I would move it under the whole `switch` block to be effective for all types of topics.

Reply:
Thanks. Fixed.